### PR TITLE
RSDK-7283 Add logs for timing of ICE events to TS dial code

### PR DESCRIPTION
**Pull request by benjirewis**
[RSDK-7283](https://viam.atlassian.net/browse/RSDK-7283)

Adds debug-level logs for:
- Gathering of local ICE candidates
- Reception of remote ICE candidates
- Statistics of caller updates for this connection establishment
     - Number of updates
     - Average duration of update
     - Max duration of update

[RSDK-7283]: https://viam.atlassian.net/browse/RSDK-7283?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: rpc/js/src/dial.ts [hunk 1]
```diff
@@ -488,10 +505,14 @@ export async function dialWebRTC(
             return;
           }
 
+          if (event.candidate.candidate !== null) {
+            console.debug(`gathered local ICE ${event.candidate.candidate}`);
```
Comment (benjirewis) [inline rpc/js/src/dial.ts:509]: In the firefox `Console` tab, this log looks like:
```
19:01:54.944 gathered local ICE candidate:5 1 UDP 92151807 34.203.251.92 22410 typ relay raddr 34.203.251.92 rport 22410
```

File: rpc/js/src/dial.ts [hunk 2]
```diff
@@ -560,6 +589,9 @@ export async function dialWebRTC(
         }
         const update = response.getUpdate()!;
         const cand = iceCandidateFromProto(update.getCandidate()!);
+        if (cand.candidate !== null) {
+          console.debug(`received remote ICE ${cand.candidate}`);
```
Comment (benjirewis) [inline rpc/js/src/dial.ts:593]: In the firefox `Console` tab, this log looks like:
```
19:01:54.952 received remote ICE candidate:2132323044 1 udp 1694498815 24.63.69.220 52236 typ srflx raddr 0.0.0.0 rport 52236
```

File: rpc/js/src/dial.ts [hunk 3]
```diff
@@ -474,6 +474,23 @@ export async function dialWebRTC(
...
+      let maxCallUpdateDuration = 0;
+      let totalCallUpdateDuration = 0;
+
+      pc.addEventListener('iceconnectionstatechange', () => {
+        if (pc.iceConnectionState !== 'completed' || numCallUpdates === 0) {
+          return;
+        }
+        let averageCallUpdateDuration =
+          totalCallUpdateDuration / numCallUpdates;
+        console.debug(
```
Comment (benjirewis) [inline rpc/js/src/dial.ts:487]: In the firefox `Console` tab, this log looks like:
```
19:01:54.951 caller update statistics: num_updates: 2, average_duration: 146ms, max_duration: 149ms
```

File: rpc/js/src/dial.ts [hunk 4]
```diff
@@ -474,6 +474,23 @@ export async function dialWebRTC(
       const offerDesc = await pc.createOffer({});
 
       let iceComplete = false;
+      let numCallUpdates = 0;
+      let maxCallUpdateDuration = 0;
+      let totalCallUpdateDuration = 0;
+
+      pc.addEventListener('iceconnectionstatechange', () => {
+        if (pc.iceConnectionState !== 'completed' || numCallUpdates === 0) {
```
Comment (benjirewis) [inline rpc/js/src/dial.ts:482]: Typescript has a similar situation to Go here: we never seem to reach `completed`, only `connected`. I've also found that in typescript, we can reach a state of `connected` without ever fully submitting a local ICE candidate to the DB (`numCallUpdates === 0`). I'm not sure it's possible to write code similar to [what we have in Go](https://github.com/viamrobotics/goutils/blob/main/rpc/wrtc_client.go#L163-L176), as I believe this TS code does not actually run in another thread and can't "wait" for more caller updates to finish.

File: rpc/js/src/dial.ts [hunk 5]
```diff
@@ -474,6 +474,23 @@ export async function dialWebRTC(
...
+          return;
+        }
+        let averageCallUpdateDuration =
+          totalCallUpdateDuration / numCallUpdates;
+        console.debug(
+          `caller update statistics: ` +
+            `num_updates: ${numCallUpdates}, ` +
+            `average_duration: ${averageCallUpdateDuration}ms, ` +
+            `max_duration: ${maxCallUpdateDuration}ms`
+        );
```
Comment (maximpertsov) [inline rpc/js/src/dial.ts:492]: until we start logging this elsewhere (sentry, etc) - let's group the output to cut down on console noise:

```suggestion
        console.groupCollapsed("Caller update statistics")
        console.table({
            num_updates: numCallUpdates,
            average_duration: `${averageCallUpdateDuration}ms`,
            max_duration: `${maxCallUpdateDuration}ms`
        });
        console.groupEnd()
```

if we don't like the table output we can just keep logs are you have them but in the collapsed group.

Comment (benjirewis) [reply rpc/js/src/dial.ts:492]: Ah yeah good idea; didn't know about groups. Couldn't take your suggestion verbatum due to lack of `;`s, but put the same change in.

**Comment by cheukt** [comment]
general question - will these always show or are they hidden?

**Comment by benjirewis** [comment]
> will these always show or are they hidden?

They will show under the `Console` tab of your browser if you enable debug logging. _How_ to do that varies a bit between browsers, but generally you'd have to open the console and explicitly enable debug.

**Review by cheukt — APPROVED** [review]
LGTM

**Review by maximpertsov — APPROVED** [review]

**Pull request merged**